src/dynamic_honey_badger/dynamic_honey_badger.rs

```diff
@@ -321,18 +327,34 @@ where
         Ok(step)
     }
 
+    pub(super) fn update_encryption_schedule(
+        &mut self,
+        epoch: u64,
+        change: &Change<N>,
+    ) -> Result<Step<C, N>> {
+        self.restart_honey_badger(epoch);
+        if let Change::EncryptionSchedule(schedule) = change {
+            self.honey_badger.encryption_schedule = *schedule;
```

Comment:
I'd prefer adding that to `restart_honey_badger` as an argument. Let's not change fields while an algorithm is already running, and add it to the `HoneyBadgerBuilder` instead.